extensions/amp-auto-ads/0.1/ad-network-config.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {DataAttributeDef} from './placement';
+import {buildUrl} from '../../../ads/google/a4a/url-builder';
+import {user} from '../../../src/log';
+
+/** @const */
+const TAG = 'amp-auto-ads';
+
+/**
+ * An interface intended to be implemented by any ad-networks wishing to support
+ * amp-auto-ads.
+ * @interface
+ */
+export class AdNetworkConfig {
+
+  /**
+   * @return {string}
+   */
+  getConfigUrl() {}
+
+  /**
+   * Any data attributes derived from either the page or the auto-amp-ads tag
+   * that should be applied to any ads inserted.
+   * @return {!Array<!DataAttributeDef>}
+   */
+  getDataAttributes() {}
+}
+
+/**
+ * Builds and returns an AdNetworkConfig instance for the given type.
+ * @param {string} type
+ * @param {!Element} autoAmpAdsElement
+ * @return {?AdNetworkConfig}
+ */
+export function getAdNetworkConfig(type, autoAmpAdsElement) {
+  if (type == 'adsense') {
+    return new AdSenseNetworkConfig(autoAmpAdsElement);
+  }
+  user().error(TAG, 'No ad network implementation for type: ' + type);
+  return null;
+}
+
+/**
+ * @implements {AdNetworkConfig}
+ */
+class AdSenseNetworkConfig {
```

Comment:
I don't know of any planned implementations beyond AdSense, so not sure how many other ad networks might implement one. Since it's not much logic per network, I feel like it's simpler and easier to read to just have one file at the moment. If you want me to split it into its own file I'm happy to do that though?